timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampTrackerImpl.java

```diff
@@ -24,16 +25,33 @@
 import java.util.SortedMap;
 import java.util.TreeMap;
 import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 import javax.annotation.concurrent.GuardedBy;
 
 final class NamedMinTimestampTrackerImpl implements NamedMinTimestampTracker {
     private final String timestampName;
+    private final AtomicInteger numLocksHeld;
```

Comment:
- Write path: this is modified in `lock` and `unlock` calls which are already synchronized
- Read path: this will be ran in some telemetry wc thread, and the read seems to be a volatile read only - I can't imagine a less perf-impacting read (other than a final field) if we are to have memory visibility. I guess we could have a variable with no synchronization and hope for the best, the metrics don't need to be 100% sure